modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/util/CommerceAccountHelperImpl.java

```diff
@@ -285,9 +288,23 @@ public AccountEntry getCurrentAccountEntry(
 				commerceChannelGroupId);
 		long userId = _portal.getUserId(httpServletRequest);
 
-		AccountEntry accountEntry =
-			_currentAccountEntryManager.getCurrentAccountEntry(
+		HttpServletRequest originalHttpServletRequest =
```

Comment:
I'm pretty sure this part could brake a lot of things.
We should only have 1 source of thruth that is the CurrentAccountEntryManager.
The CurrentAccountEntryManager will use the CurrentAccountEntryManagerStore to get it from the session...

Reply:
I will re-investigate the ramifications as well as exploring a better solution. I believe I have a better idea to address your concern here.